android/src/test/java/com/onegini/mobile/sdk/reactnative/AuthenticateUserImplicitlyUseCaseTests.kt

```diff
@@ -0,0 +1,113 @@
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onegini.mobile.sdk.reactnative.clean.use_cases.AuthenticateUserImplicitlyUseCase
+import com.onegini.mobile.sdk.reactnative.exception.OneginiWrapperErrors.PROFILE_DOES_NOT_EXIST
+import com.onegini.mobile.sdk.reactnative.managers.AuthenticatorManager
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Answers
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@RunWith(MockitoJUnitRunner::class)
+class AuthenticateUserImplicitlyUseCaseTests {
+
+    @get:Rule
+    val reactArgumentsTestRule = ReactArgumentsTestRule()
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    lateinit var oneginiSdk: OneginiSDK
+
+    @Mock
+    lateinit var promiseMock: Promise
+
+    @Mock
+    lateinit var oneginiImplicitTokenRequestError: OneginiImplicitTokenRequestError
+
+    lateinit var authenticateUserImplicitlyUseCase: AuthenticateUserImplicitlyUseCase
+
+    lateinit var authenticatorManager: AuthenticatorManager
+
+    private val nonExistingProfileId = "nonExistingProfileId"
+    private val existingUserProfileId = "123456"
+
+    @Before
+    fun setup() {
+        authenticatorManager = AuthenticatorManager(oneginiSdk)
+        authenticateUserImplicitlyUseCase = AuthenticateUserImplicitlyUseCase(oneginiSdk, authenticatorManager)
+    }
+
+    @Test
+    fun `When supplying a profileId that does not exist, Then it should reject with PROFILE_DOES_NOT_EXIST`() {
+        whenNoRegisteredProfiles()
+        authenticateUserImplicitlyUseCase(nonExistingProfileId, null, promiseMock)
+        verify(promiseMock).reject(PROFILE_DOES_NOT_EXIST.code.toString(), PROFILE_DOES_NOT_EXIST.message)
+    }
+
+    @Test
+    fun `When supplying a profileId that does exist, Then it should call authenticateUserImplicitly on userClient with the correct profile`() {
+        val userProfile = UserProfile(existingUserProfileId)
+        whenRegisteredProfile(userProfile)
+        authenticateUserImplicitlyUseCase(existingUserProfileId, null, promiseMock)
+        verify(oneginiSdk.oneginiClient.userClient).authenticateUserImplicitly(eq(userProfile), any(), any())
+    }
+
+    @Test
+    fun `When supplying a profileId that does exist and the handler calls onSuccess, Then the promise should resolve with null`() {
+        val userProfile = UserProfile(existingUserProfileId)
+        whenRegisteredProfile(userProfile)
+        whenever(oneginiSdk.oneginiClient.userClient.authenticateUserImplicitly(any(), any(), any())).thenAnswer {
+            it.getArgument<OneginiImplicitAuthenticationHandler>(2).onSuccess(userProfile)
```

Comment:
what does 2 stands for? 🤔

Reply:
That's the index of the argument to get. So the third one in this case on this function. 
```
  public void authenticateUserImplicitly(@NonNull UserProfile userProfile, @Nullable String[] scopes, @NonNull OneginiImplicitAuthenticationHandler implicitAuthenticationHandler) {
```


``` 
   /**
     * Returns casted argument at the given index.
     *
     * Can lookup in expanded arguments form {@link #getArguments()}.
     *
     * This method is preferred over {@link #getArgument(int, Class)} for readability. Please read
     * the documentation of {@link #getArgument(int, Class)} for an overview of situations when
     * that method is preferred over this one.
     *
     * @param index argument index
     * @return casted argument at the given index
     * @since 2.1.0
     */
    <T> T getArgument(int index);
```